workers.go

```diff
@@ -0,0 +1,259 @@
+	Pattern string `json:"pattern"`
+	Enabled bool   `json:"enabled"`
+}
+
+// WorkerRoutesResponse embeds Response struct and slice of WorkerRoutes
+type WorkerRoutesResponse struct {
+	Response
+	Routes []WorkerRoute `json:"result"`
+}
+
+// WorkerRouteResponse embeds Response struct and a single WorkerRoute
+type WorkerRouteResponse struct {
+	Response
+	WorkerRoute `json:"result"`
+}
+
+// WorkerScript Cloudflare Worker struct with metadata
+type WorkerScript struct {
+	WorkerMetaData
+	Script string `json:"script"`
+}
+
+// WorkerMetaData contains worker script information such as size, creation & modification dates
+type WorkerMetaData struct {
+	ID         string    `json:"id,omitempty"`
+	ETAG       string    `json:"etag,omitempty"`
+	Size       int       `json:"size,omitempty"`
+	CreatedOn  time.Time `json:"created_on,omitempty"`
+	ModifiedOn time.Time `json:"modified_on,omitempty"`
+}
+
+// WorkerListResponse wrapper struct for API response to worker script list API call
+type WorkerListResponse struct {
+	Response
+	WorkerList []WorkerMetaData `json:"result"`
+}
+
+// WorkerScriptResponse wrapper struct for API response to worker script calls
+type WorkerScriptResponse struct {
+	Response
+	WorkerScript `json:"result"`
+}
+
+// DeleteWorker deletes worker for a zone.
+//
+// API reference: https://api.cloudflare.com/#worker-script-delete-worker
+func (api *API) DeleteWorker(zoneID string) (WorkerScriptResponse, error) {
+	uri := "/zones/" + zoneID + "/workers/script"
+	res, err := api.makeRequest("DELETE", uri, nil)
+	var r WorkerScriptResponse
+	if err != nil {
+		return r, errors.Wrap(err, errMakeRequestError)
+	}
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return r, errors.Wrap(err, errUnmarshalError)
+	}
+	return r, nil
+}
+
+// DeleteWorkerWithName deletes worker for a zone.
+// This is an enterprise only feature https://developers.cloudflare.com/workers/api/config-api-for-enterprise
+// organizationID must be specified as api option https://godoc.org/github.com/cloudflare/cloudflare-go#UsingOrganization
+//
+// API reference: https://api.cloudflare.com/#worker-script-delete-worker
+func (api *API) DeleteWorkerWithName(scriptName string) (WorkerScriptResponse, error) {
+	if api.organizationID == "" {
+		return WorkerScriptResponse{}, errors.New("organization ID required for enterprise only request")
+	}
+	uri := "/accounts/" + api.organizationID + "/workers/scripts/" + scriptName
+	res, err := api.makeRequest("DELETE", uri, nil)
+	var r WorkerScriptResponse
+	if err != nil {
+		return r, errors.Wrap(err, errMakeRequestError)
+	}
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return r, errors.Wrap(err, errUnmarshalError)
+	}
+	return r, nil
+}
+
+// DownloadWorker fetch raw script content for your worker returns []byte containing worker code js
+//
+// API reference: https://api.cloudflare.com/#worker-script-download-worker
+func (api *API) DownloadWorker(zoneID string) (WorkerScriptResponse, error) {
```

Comment:
To expand on my suggestion here:

`DownloadWorker(zoneID string, conf *WorkerConf) (WorkerScriptResponse, error)`

`*WorkerConf` can be something like -

```
type WorkerConf struct {
    ScriptName string
    // Future options here
}
```

This avoids having double the methods, which also has potential to change in the future if Cloudflare opened up multi-script/supported it as the default. If a non `nil` `*WorkerConf` is provided, use the name in there.

Keeping the API surface small, but making the default case "easy" -- which is single-script, zoneID based today - should strike a good balance.